### PR TITLE
re-name & re-organize files

### PR DESCRIPTION
**Pull request by alvinzz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Reference sparse coding implementations for efficient learning and inference imp
 
 ## Implemented Inference Methods
 * Locally Competative Algorithm (LCA)
-* Gradient Descent with Euler's method on Laplace Prior (Vanilla)
+* Gradient Descent with Euler's method on Laplace Prior (Euler)
```
Comment (belsten) [inline README.md:9]: Curious about the rename. While Vanilla is a bit obscure, it encompasses the method (euler), objective (laplace), and historical context (original 1996 paper). Euler is almost more vague. Many inference methods could use Euler's method. Thoughts on keeping this inference method called vanilla?

**Review by belsten — COMMENTED** [review]
Consistency? If we're changing this inference method to euler then the test needs to be re-written amongst other things. All other test cases are capitalized too. 
This is a big change.

**Review by belsten — COMMENTED** [review]
data should remain in the sparsecoding directory so users can import it via 
`sparsecoding.data`

File: README.md (hunk 1 above)
Comment (alvinzz) [reply README.md:9]: Maybe `Nature96` or `Olshausen96`

**Comment by alvinzz** [comment]
Discussed in-person, decided to leave `data` in `sparsecoding`. `Vanilla` renaming would be a separate PR.

**Pull request closed without merging**